### PR TITLE
Reproduction

### PR DESCRIPTION
**Pull request by MylesScholz**
Changes:
- Reproduction
    - Nucleus::activate() sends a signal to CellSpawner if the Cell's resources meet both thresholds and are sufficient
    - Nucleus::activate() subtracts the cost of reproduction and halves the parent Cell's remaining resources
    - CellSpawner instantiates a new Cell, uses Cell::seteq() make it a clone of its parent, and adds it to the CellEnvironment with some initial movement
    - Cell::seteq() clones the Genome of the parent Cell and gives it to the childCell
    - Cell::seteq() copies relevant attributes from the parent Cell to the child Cell, resets both Cells' age and birth time, and rescales both Cells to half the parent Cell's original area each
- Cell fixes and improvements
    - applyScale() prevents resource overflow and updates _spriteSize
    - Genes are expressed in _ready(), and the resulting CellStructures are scaled properly
    - Homeostasis and reproduction costs are set in _ready() after Genome expression
- Infinite resources fix
    - Replaced _conversionRate with _preferredConversionRate in Mitochondria and Ribosomes classes
    - Added _nutrientEnergyConversionRate to CellState, which is used by both Mitochondria and Ribosomes
    - If _nutrientEnergyConversionRate is unset when either Mitochondria or Ribosomes activate, it will be set to the respective _preferredConversionRate value
- Added _resourceProportion to CellSpawner--the proportion of nutrients and energy that Cells spawn with
- Gene cloning
- NutrientZone negative feeding and applyScale fixes
- Balance tweaks
- Documentation updates

### CODE REVIEW

File: docs/src/cell.md [hunk 1]
```diff
@@ -30,7 +32,7 @@ Description: The Cell object represents a biological cell within a simulation. I
 
 - void setImmortal(bool isImmortal): sets the global variable immortal to the given value
 
-- void _ready() override: runs once when the Cell has been created and is ready to be used; points _cellState to this Cell's CellState Node
+- void _ready() override: runs once when the Cell has been created and is ready to be used; points _cellState to this Cell's CellState Node; sets homeostasis and reproduction costs
```
Comment (Alex-x90) [inline docs/src/cell.md:35]: One thing to note is that `_ready()` happens when an object is added to the node tree (using `add_child()` for example), not when it's created.

File: docs/src/genome.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# Genome Description
+
+## Classes
+
+Genome
+Inherits: none
+Inherited by: none
+
+Description: the Genome is class representing the genetic material of a cell; it contains and manages the Cell's Genes
```
Comment (Alex-x90) [inline docs/src/genome.md:9]: The wording is a little messy here

File: src/cell.cpp [hunk 3]
```diff
@@ -56,7 +90,10 @@ Cell::Cell() {
 	// Instantiate the random number generator
 	_rand.instantiate();
 
-	//temp setup a genome for testing.
+	// Instantiate Genome
+	_cellGenome = Genome();
```
Comment (Alex-x90) [inline src/cell.cpp:94]: Not necessary as the `Genome _cellGenome;` definition from the header will instantiate an instance of the genome class. It's only done like this inside seteq in order to deconstruct/overwrite the genes added by the default constructor.

File: src/cell_spawner.cpp [hunk 4]
```diff
@@ -107,10 +125,39 @@ void CellSpawner::spawnCell(bool isImmortal) {
 	cellEnvironment->add_child(cell);
 	cell->connect("cell_death", Callable(cellEnvironment, "_on_cell_death"));
 
+	if (!isImmortal) // Do we want the cells on the title screen to be able to reproduce?
```
Comment (Alex-x90) [inline src/cell_spawner.cpp:128]: Should remove this comment now

File: src/cell_spawner.cpp [hunk 5]
```diff
@@ -107,10 +125,39 @@ void CellSpawner::spawnCell(bool isImmortal) {
...
+	RandomNumberGenerator rand = RandomNumberGenerator();
+
+	Node *childCell = _cellScene->instantiate();
+	Cell *cellObject = Object::cast_to<Cell>(childCell);
+
+	cellObject->seteq(cell);
+
+	cellObject->set_position(cell->get_position());
+
+	// This probably shouldn't be random for a child cell, and I assume w/ the flagella rework it won't be
```
Comment (Alex-x90) [inline src/cell_spawner.cpp:146]: Is this still relevant?

File: src/nucleus.cpp [hunk 6]
```diff
@@ -17,8 +23,20 @@ void Nucleus::activate(CellState *cellState) {
 	if (this->getSprite()->get_frame() == this->getSprite()->get_sprite_frames()->get_frame_count("activate") - 1)
 		this->getSprite()->stop();
 
-	if (cellState->getTotalNutrients() >= _reproductionNutrientThreshold) {
-		// UtilityFunctions::print("Nucleus activated: ", cellState->getTotalNutrients(), " >= ", _reproductionNutrientThreshold);
+	// Check that total resources are greater than or equal to the activation threshold and the reproduction cost
+	bool thresholdCondition = cellState->getTotalNutrients() >= _reproductionNutrientThreshold * cellState->getNutrientMaximum() && cellState->getTotalNutrients() >= cellState->getReproductionNutrientCost();
+	thresholdCondition = thresholdCondition && cellState->getTotalEnergy() >= _reproductionEnergyThreshold * cellState->getEnergyMaximum() && cellState->getTotalEnergy() >= cellState->getReproductionEnergyCost();
+
```
Comment (Alex-x90) [inline src/nucleus.cpp:29]: Nitpick (feel free to ignore): Could instead define as nutrient & energy threshold conditions, then and them together in the if statement below which would be a bit clearer.

File: src/cell_spawner.cpp (hunk 5 above)
Comment (MylesScholz) [reply src/cell_spawner.cpp:146]: We could move it to seteq().

File: docs/src/genome.md (hunk 2 above)
Comment (MylesScholz) [reply docs/src/genome.md:9]: Better?

**Review by WyattHolliday — APPROVED** [review]

**Pull request merged**